### ISSUE TITLE
[BUG] 'PointerEvent' can't be assigned to the parameter type 'PointerDownEvent'

### ISSUE THREAD

**Issue by zs-dima** [fireslime/photo_view#441]
```
Flutter 2.4.0-0.0.pre • channel dev
Framework • revision 96bbcd006f (8 days ago) • 2021-06-30 03:43:41 -0400
Engine • revision c633b2a6a0
Tools • Dart 2.14.0 (build 2.14.0-258.0.dev)
```

Packages installed
native_pdf_view

Error:
```
photo_view-0.11.1/lib/src/core/photo_view_gesture_detector.dart:106:29: Error: The argument type 'PointerEvent' can't be assigned to the parameter type 'PointerDownEvent'.
 - 'PointerEvent' is from 'package:flutter/src/gestures/events.dart' ('/C:/dev/flutter/packages/flutter/lib/src/gestures/events.dart').
 - 'PointerDownEvent' is from 'package:flutter/src/gestures/events.dart' ('/C:/dev/flutter/packages/flutter/lib/src/gestures/events.dart').
    super.addAllowedPointer(event);
```

**Comment by radvansky-tomas** [comment]
@zs-dima change channel do STABLE

**Comment by zs-dima** [comment]
@radvansky-tomas thanks for reply.
I am using Dev channel for this project.

**Comment by radvansky-tomas** [comment]
I understand, but you cannot expect plugins to move to DEV, fork it...fix that line and maybe send PR back so its prepared when it makes to stable version

**Comment by zs-dima** [comment]
@radvansky-tomas but all plugin working well with Dev channel except this one, that quite unusual

**Comment by mayder** [comment]
https://flutter.dev/docs/release/breaking-changes/gesture-recognizer-add-allowed-pointer

**Comment by Vinzent03** [comment]
Does the fix work on stable too? If so, I would love to see the pr merged, because the breaking change is on beta channel already too.

**Comment by SDAChess** [comment]
> Does the fix work on stable too? If so, I would love to see the pr merged, because the breaking change is on beta channel already too.

I tested my fix on stable and it works. I don't think the PR is going to get merged any time soon, because I'm not sure how much time the author of the library has. Still if you need the fix you can take my branch as reference for the library code with something like this in your pubspec : 
```
photo_view:
       git:
         url: git://github.com/SDAChess/photo_view.git
         ref: SDAChess-patch-441
```
Not quite sure it would work though but still a good thing to try.

**Comment by shengdoushi** [comment]
> > Does the fix work on stable too? If so, I would love to see the pr merged, because the breaking change is on beta channel already too.
> 
> I tested my fix on stable and it works. I don't think the PR is going to get merged any time soon, because I'm not sure how much often the author releases a new version. Still if you need the fix you can take my branch as reference for the library code with something like this in your pubspec :
> 
> ```
> photo_view:
>        git:
>          url: git@github.com:SDAChess/photo_view.git
>          ref: SDAChess-patch-441
> ```
> 
> Not quite sure it would work though but still a good thing to try.

Some library using photo_view's some version, how to force use git's version?

**Comment by jseminck** [comment]
@renancaraujo, sorry for pinging you but is there any chance we can get the PR merged? The latest flutter channel branch contains much needed performance improvements (shader compilations) and we have some dependencies that rely on this package. Thanks so much for your work!

**Comment by renancaraujo** [comment]
Hello people, 

I just woke up from a coma and created a PR that addresses this problem. 
https://github.com/fireslime/photo_view/pull/445 should work for stable, beta, dev, and master channels.

I will do some testing and it should be included in the next release.

**Comment by renancaraujo** [comment]
This should be fixed on 0.12.0:  https://github.com/fireslime/photo_view/releases/tag/0.12.0

**Comment by flutterq** [comment]
[[Already Solved] The argument type ‘PointerEvent’ can’t be assigned to the parameter type ‘PointerDownEvent’](https://fluttercorner.com/the-argument-type-pointerevent-cant-be-assigned-to-the-parameter-type-pointerdownevent/)
